eth2/simulation/src/simulation.rs

```diff
@@ -0,0 +1,88 @@
+use crate::errors::{
+    Result,
+    WhatBound,
+};
+use crate::store::Store;
+use types::eth_spec::{EthSpec};
+use types::slot_epoch_root::{
+    EEIndex,
+    Root,
+    Shard,
+    ShardSlot,
+};
+use types::execution_environment::ExecutionEnvironment;
+use types::shard_block::ShardBlock;
+use types::shard_state::ShardState;
+use types::shard_transaction::ShardTransaction;
+
+pub struct Simulation<T>
+where
+    T: EthSpec,
+{
+    store: Store<T>,
+}
+
+impl<T: EthSpec> Simulation<T> {
+    /// Add a new execution environment, return EE index
+    fn create_execution_environment(a: args::CreateExecutionEnvironment) -> EEIndex {
+        unimplemented!();
```

Comment:
Would this method expect wasm code and an initial state?

Reply:
Hmm certainly it would need to somehow be able to generate initial state, if not have it passed in explicitly.  For now I'll add the ability to pass it in explicitly, we can update later if the reality is slightly different.